src/optimizer/plugins/RemoveGeneratedTextRulesOptimizer.ts

```diff
@@ -0,0 +1,36 @@
+import { OptimizerPlugin } from '../OptimizerPlugin';
+import { Package } from '../../processor';
+import { ExportableCaretValueRule, ExportableRule } from '../../exportable';
+import { pullAt } from 'lodash';
+import { fshtypes } from 'fsh-sushi';
+
+export default {
+  name: 'remove_generated_text_rules',
+  description: 'Remove caret rules on "text" which represent generated content.',
+
+  optimize(pkg: Package): void {
+    [...pkg.profiles, ...pkg.extensions, ...pkg.valueSets, ...pkg.codeSystems].forEach(resource => {
+      const rulesToRemove: number[] = [];
+      const hasGeneratedText = resource.rules.some(
+        (rule: ExportableRule) =>
+          rule instanceof ExportableCaretValueRule &&
+          rule.path === '' &&
+          rule.caretPath === 'text.status' &&
+          rule.value instanceof fshtypes.FshCode &&
+          ['generated', 'extensions'].includes(rule.value.code)
+      );
+      if (hasGeneratedText) {
+        resource.rules.forEach((rule: ExportableRule, i: number) => {
+          if (
+            rule instanceof ExportableCaretValueRule &&
+            rule.path === '' &&
+            rule.caretPath.match(/text\./)
```

Comment:
Should this match rules that have `text.` anywhere in their `caretPath`, or only those where the path starts with `text.`?

Reply:
Good question, @mint-thompson.  I feel like we probably really want `/^text\./`  I can't think of any cases where this would be embedded in a deeper path -- and if it were, I'm not sure we'd want to clear it anyway.

Reply:
Ah yikes, thanks for finding this. It's especially bad, since there are several elements on StructureDefinitions that would contain "text." in the path. Fixed and tested in https://github.com/FHIR/GoFSH/pull/30/commits/395fd59b072369e2623c603280a6e0c8a41aae32.